charts/bctlquickstart/templates/agent/bctl-agent-role.yaml

```diff
@@ -8,7 +8,7 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
-  namespace: {{ include "bctlquickstartchart.namespace" . }}
+  namespace: {{ .Release.Namespace }}
 rules:
     - apiGroups: [""]
       resources: ["secrets"]
```

Comment:
As you mentioned. lets add `resourceNames:` to limit the scope even more.

Reply:
This change will take more time than expected. We can't just update the role because we need the `create` permission as the agent currently creates the secret in the cluster on init for the first time.

However, we can't use `create` with `resourceNames`:

>  Note: You cannot restrict **create** or deletecollection requests by their resource name. For create, this limitation is because the name of the new object may not be known at authorization time. If you restrict list or watch by resourceName, clients must include a metadata.name field selector in their list or watch request that matches the specified resourceName in order to be authorized. For example, kubectl get configmaps --field-selector=metadata.name=my-configmap

Reference: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources

---

If we want this change, I think it will require agent changes + chart changes + get agent yaml changes (on backend).

Possible new design:
- Generate kube yaml (on backend) + Helm chart both create the secret (`bctl-agent-secret`). But keep the secret empty
- Agent vault code is changed to update the secret by name instead of creating it. Basically, the agent should assume it exists as both of our installation methods will have already created empty versions of them.

Then, since the agent only needs to update the secret, we can use the `resourceName` restriction on this specific secret

Reply:
Made a ticket here: https://commonwealthcrypto.atlassian.net/browse/CWC-1564